src/studio/src/designer/backend/Controllers/RepositoryController.cs

```diff
@@ -354,6 +355,20 @@ public async Task<RepositoryModel> CreateApp(string org, string repository)
             return await _repository.CreateService(org, config);
         }
 
+        /// <summary>
+        /// Action used to copye an existing app under the current org.
+        /// </summary>
+        /// <returns>
+        /// The newly created repository.
+        /// </returns>
+        [Authorize]
+        [HttpPost]
+        public async Task<RepositoryModel> CopyApp(string org, string sourceRepository, string targetRepository)
+        {
+            string developer = AuthenticationHelper.GetDeveloperUserName(HttpContext);
+            return await _repository.CopyRepository(org, sourceRepository, targetRepository, developer);
```

Comment:
Since it's a POST we should return 201 Created.

Reply:
the repositorymodel contains a property for the HttpStatus of the creation of the repository. Returning 201 in all cases might be misleading. I guess we could map the most common creationStatuses to a response. Either way the behavior of the endpoints CreateApp and CopyApp should be the same if we decide to change it

Reply:
Yes, I noticed. But my idea was that it should only return 201 Created if it's successful. Otherwise it could return the appropriate error status from the repositorymodel if something has gone wrong. That way we're allowing the client to respond the http status directly instead of parsing an object to find the status.